homeassistant/components/openai_conversation/conversation.py

```diff
@@ -39,6 +45,15 @@ async def async_setup_entry(
     async_add_entities([agent])
 
 
+def _format_tool(tool: llm.Tool) -> dict[str, Any]:
+    """Format tool specification."""
+    tool_spec = {"name": tool.name}
+    if tool.description:
+        tool_spec["description"] = tool.description
+    tool_spec["parameters"] = convert(tool.parameters)
```

Comment:
As noted in another comment, the request will fail if `tool_spec["parameters"]["properties"]` does not exist.

Reply:
Solved in voluptuous-openapi==0.0.4